src/build/mdn-comments.ts

```diff
@@ -47,66 +45,109 @@ function extractSummary(markdown: string): string {
     return sentenceMatch[0]; // Return the first full sentence
   }
 
-  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+  const firstWord = normalizedText.split(" ")[0];
+  return firstWord || "";
 }
 
-async function getDirectories(dirPath: URL): Promise<URL[]> {
-  try {
-    const entries = await fs.readdir(dirPath, {
-      withFileTypes: true,
-    });
-    return entries
-      .filter((entry) => entry.isDirectory())
-      .map((entry) => new URL(entry.name + "/", dirPath));
-  } catch (error) {
-    console.error("Error reading directories:", error);
-    return [];
+async function walkDirectory(dir: URL): Promise<URL[]> {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
+  const parentDirName = dir.pathname.split("/").filter(Boolean).slice(-1)[0];
+  let results: URL[] = [];
+
+  for (const entry of entries) {
+    if (entry.isDirectory()) {
+      if (entry.name === parentDirName) continue;
+      const subDir = new URL(`${entry.name}/`, dir);
+      results = results.concat(await walkDirectory(subDir));
+    } else if (entry.isFile() && entry.name === "index.md") {
+      results.push(new URL(entry.name, dir));
+    }
   }
+
+  return results;
 }
 
-async function getIndexMdContents(
-  folders: URL[],
-): Promise<{ [key: string]: string }> {
-  const results: { [key: string]: string } = {};
+const paths: Record<string, string[]> = {
+  "web-api-instance-property": ["properties", "property"],
+  "web-api-static-property": ["properties", "property"],
+  "web-api-instance-method": ["methods", "method"],
+  "web-api-static-method": ["methods", "method"],
+  "web-api-interface": [],
+  "webgl-extension": [],
+  "webgl-extension-method": ["methods", "method"],
+};
 
-  for (const folder of folders) {
-    const indexPath = new URL("index.md", folder);
+function generateTypes(content: string): string[] | undefined {
+  const pageType = content.match(/page-type:\s*["']?([^"'\n]+)["']?/)!;
+  const type = pageType[1];
+  const plural = paths[type];
+  if (!plural) return;
```

Comment:
You can just return `paths[type]` directly. Also given this return a "path", would be nice to name the function `generatePath`.